### PR TITLE
Organize table - toggle selection

### LINKED ISSUE #8579: Add a Toggle Selection Button in the Organize columns window (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#54930
Author: @Poulpator
QGIS version: 3.34

**Add a Toggle Selection Button in the Organize columns window**

## Description

Add a Toggle Selection Button in the Organize columns window

This is a little quality of life improvement  to manage datasets with a growing number of fields.

How it work, it use the classic multiple selection with shift or ctrl to select multiple : 
Note the Toggle Selection as well as the Deselect or select All only appear when there is more than 5 fields

Screencast from 12-10-2023 23 03 54

Motivation for this change: The number of fields can quickly grow as we join multiple sources of data/database tables. For example the National Building Database from France, the export in gpkg format have 283 fields. This change make it more easy to manage what's being displayed in the attribute table

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by selmaVH1**
fixes #8579

### CODE REVIEW

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 1]
```diff
@@ -204,17 +204,28 @@ from the attribute table (in "table view" mode).
...
+  actively restored
 * drag-and-drop items to reorder the columns in the attribute table. Note that
   this change is for the table rendering and does not alter the fields order in
   the layer datasource
 * add a new virtual :guilabel:`Actions` column that displays in each row a
   drop-down box or a button list of enabled actions.
-  See :ref:`actions_menu` for more information about actions.
+  See :ref:`actions_menu` for more information about actions
+* choose :guilabel:`Select All` to select all columns or 
+* :guilabel:`Deselect All` to deselect all columns,
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/attribute_table.rst:227]: Actually, these buttons do not **select** (in the meaning, selecting something in a list); they show or hide columns, they check or uncheck visibility of all the fields. And in this way, I would see their mention close to the first point "check/uncheck...." as a way to _batch_ do it.
And maybe should we have a bullet point on how people can select multiple fields in the list (as far as I can see, it works like in the main attribute table, with keys combination)
Note that I suggest to rename these buttons (https://github.com/qgis/QGIS/pull/56906)

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 2]
```diff
@@ -204,17 +204,28 @@ from the attribute table (in "table view" mode).
...
+* use the :guilabel:`Toggle selection` to invert the current selection of columns.
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/attribute_table.rst:228]: how about
```suggestion
* use :guilabel:`Toggle selection` to invert visibility of the current selection of columns.
```
?

File: docs/user_manual/working_with_vector/img/organize_table_columns.png
Comment (DelazJ) [inline docs/user_manual/working_with_vector/img/organize_table_columns.png:1]: Is there a way to not show mouse pointer on the screenshots?

File: docs/user_manual/working_with_vector/attribute_table.rst (hunk 1 above)
Comment (DelazJ) [reply docs/user_manual/working_with_vector/attribute_table.rst:227]: The pr is merged; we can already adjust the buttons name. And as the screenshot needs update, I wonder if we really need a screenshot here (there are no real options in this dialog)

Comment (selmaVH1) [reply docs/user_manual/working_with_vector/attribute_table.rst:227]: Noted.

**Pull request merged**